data_steward/tools/clean_project_datasets.py

```diff
@@ -8,68 +14,79 @@
 
 # Project imports
 from utils import bq
+from utils import pipeline_logging
 
-logging.basicConfig(
-    stream=sys.stdout,
-    level=logging.INFO,
-    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s")
-
+LOGGER = logging.getLogger(__name__)
 
-def get_datasets_with_substrings(datasets_list, name_substrings):
-    """
-    Filters list of datasets with specified substrings (e.g. github usernames) in them
 
-    :param datasets_list: list of dataset_ids
-    :param name_substrings: identifies substrings that help identify datasets to delete
-    :return: list of dataset_ids with any substring in their dataset_id
-    """
-    datasets_with_substrings = []
-    for dataset in datasets_list:
-        if any(name_substring in dataset for name_substring in name_substrings):
-            datasets_with_substrings.append(dataset)
-    return datasets_with_substrings
-
-
-def delete_datasets(project_id, datasets_to_delete_list):
+def delete_datasets(client, datasets_to_delete_list, test_env=True):
     """
     Deletes datasets using their dataset_ids
 
-    :param project_id: identifies the project
+    :param client: client object associated with project to delete datasets from
     :param datasets_to_delete_list: list of dataset_ids to delete
+    :param offboarding_dev: boolean set to true to ensure if you are deleting 
+        datasets outside of test environment, you really intend to do that
     :return:
     """
+    # a fail safe to prevent deleting datasets in environments other than test.
+    if 'test' not in client.project and test_env:
+        raise RuntimeError(
+            "Attempting to delete datasets out of non-test environment "
+            "while not explicitly making this choice!!")
```

Comment:
I considered that and actually prefer it.  I just moved the check close to where we are actually deleting datasets to prevent someone from importing just this function and skipping the check if it was in a previous function.  I'm fine with moving this forward in the script if you're agreeable to making this function "private" so someone would have to intentionally name mangle the import to get to it.

Reply:
Thanks, I hadn't considered that. Sounds good to me.